examples/nomad-consul-ami/nomad-consul-docker-ecr.json

```diff
@@ -0,0 +1,159 @@
+        "filters": {
+          "virtualization-type": "hvm",
+          "architecture": "x86_64",
+          "name": "*ubuntu-xenial-16.04-amd64-server-*",
+          "block-device-mapping.volume-type": "gp2",
+          "root-device-type": "ebs"
+        },
+        "owners": [
+          "099720109477"
+        ],
+        "most_recent": true
+      },
+      "ssh_username": "ubuntu"
+    },
+    {
+      "ami_name": "nomad-consul-amazon-linux-{{isotime | clean_ami_name}}",
+      "ami_description": "An Amazon Linux AMI that has Nomad and Consul installed.",
+      "instance_type": "t2.micro",
+      "name": "amazon-linux-ami",
+      "region": "{{user `aws_region`}}",
+      "profile": "{{ user `aws_profile`}}",
+      "type": "amazon-ebs",
+      "source_ami_filter": {
+        "filters": {
+          "virtualization-type": "hvm",
+          "architecture": "x86_64",
+          "name": "*amzn-ami-hvm-*",
+          "block-device-mapping.volume-type": "gp2",
+          "root-device-type": "ebs"
+        },
+        "owners": [
+          "amazon"
+        ],
+        "most_recent": true
+      },
+      "ssh_username": "ec2-user"
+    }
+  ],
+  "provisioners": [
+    {
+      "type": "file",
+      "source": "setup_ubuntu16.sh",
+      "destination": "/tmp/",
+      "only": [
+        "ubuntu16-ami"
+      ]
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "/tmp/setup_ubuntu16.sh"
+      ],
+      "only": [
+        "ubuntu16-ami"
+      ]
+    },
+    {
+      "type": "file",
+      "source": "setup_amazon-linux.sh",
+      "destination": "/tmp/",
+      "only": [
+        "amazon-linux-ami"
+      ]
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "/tmp/setup_amazon-linux.sh"
+      ],
+      "only": [
+        "amazon-linux-ami"
+      ]
+    },
+    {
+      "type": "file",
+      "source": "setup_ecr_helper.sh",
+      "destination": "/tmp/"
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "/tmp/setup_ecr_helper.sh"
+      ]
+    },
+    {
+      "type": "file",
+      "source": "{{template_dir}}/../../../terraform-aws-nomad",
+      "destination": "/tmp",
+      "pause_before": "30s"
+    },
+    {
+      "type": "file",
+      "source": "setup_nomad_consul.sh",
+      "destination": "/tmp/"
+    },
+    {
+      "type": "shell",
+      "environment_vars": [
+        "NOMAD_VERSION={{user `nomad_version`}}",
+        "CONSUL_VERSION={{user `consul_version`}}",
+        "CONSUL_MODULE_VERSION={{user `consul_module_version`}}"
+      ],
+      "inline": [
+        "/tmp/setup_nomad_consul.sh"
+      ]
+    },
+    {
+      "type": "file",
+      "source": "aws_credentials",
+      "destination": "/home/ubuntu/.aws/credentials",
+      "only": [
+        "ubuntu16-ami"
+      ]
+    },
+    {
+      "type": "file",
+      "source": "aws_credentials",
+      "destination": "/home/ec2-user/.aws/credentials",
+      "only": [
+        "amazon-linux-ami"
+      ]
+    },
```

Comment:
Per my comment above, this should be done using an IAM role at run time rather than hard-coded, plaintext credentials at build time.